### PR TITLE
Polish @Value class inspections

### PR DESCRIPTION
**Pull request by wyhasany**
1. static fields should not be affected with these inspections
2. final modifier on @Value class should be marked as redundant

That's an extension of #749 PR
So they should mark following elements:
image

### CODE REVIEW

**Comment by mplushnikov** [comment]
# [Codecov](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=h1) Report
> Merging [#750](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=desc) into [master](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/commit/c0df8e56cf428b503d57b6805113a07dc43b28e2&el=desc) will **decrease** coverage by `0.03%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=tree)

```diff
@@             Coverage Diff              @@
##             master     #750      +/-   ##
============================================
- Coverage     70.00%   69.96%   -0.04%     
+ Complexity     1803     1800       -3     
============================================
  Files           201      200       -1     
  Lines          6251     6256       +5     
  Branches       1271     1273       +2     
============================================
+ Hits           4376     4377       +1     
- Misses         1265     1266       +1     
- Partials        610      613       +3     
```


| [Impacted Files](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=tree) | Coverage Δ | Complexity Δ | |
|---|---|---|---|
| [.../intellij/plugin/psi/LombokLightMethodBuilder.java](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9kZS9wbHVzaG5pa292L2ludGVsbGlqL3BsdWdpbi9wc2kvTG9tYm9rTGlnaHRNZXRob2RCdWlsZGVyLmphdmE=) | `42.37% <0.00%> (-1.70%)` | `26.00% <0.00%> (-1.00%)` | |
| [...llij/plugin/action/lombok/LombokLoggerHandler.java](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9kZS9wbHVzaG5pa292L2ludGVsbGlqL3BsdWdpbi9hY3Rpb24vbG9tYm9rL0xvbWJva0xvZ2dlckhhbmRsZXIuamF2YQ==) | `90.90% <0.00%> (-0.27%)` | `16.00% <0.00%> (ø%)` | |
| [...llij/plugin/provider/LombokInspectionProvider.java](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9kZS9wbHVzaG5pa292L2ludGVsbGlqL3BsdWdpbi9wcm92aWRlci9Mb21ib2tJbnNwZWN0aW9uUHJvdmlkZXIuamF2YQ==) | `50.00% <0.00%> (ø)` | `2.00% <0.00%> (ø%)` | :arrow_up: |
| [...j/plugin/extension/LombokElementRenameHandler.java](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9kZS9wbHVzaG5pa292L2ludGVsbGlqL3BsdWdpbi9leHRlbnNpb24vTG9tYm9rRWxlbWVudFJlbmFtZUhhbmRsZXIuamF2YQ==) | `0.00% <0.00%> (ø)` | `0.00% <0.00%> (ø%)` | :arrow_up: |
| [...j/plugin/language/LombokConfigFileTypeFactory.java](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9kZS9wbHVzaG5pa292L2ludGVsbGlqL3BsdWdpbi9sYW5ndWFnZS9Mb21ib2tDb25maWdGaWxlVHlwZUZhY3RvcnkuamF2YQ==) | | | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=footer). Last update [c0df8e5...6d4fb06](https://codecov.io/gh/mplushnikov/lombok-intellij-plugin/pull/750?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 1]
```diff
@@ -56,16 +55,35 @@ public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder holder, fina
...
+
+      if (clazz.hasAnnotation(lombok.Value.class.getName())) {
+        final PsiModifierList modifierList = clazz.getModifierList();
+        if (modifierList != null && modifierList.hasExplicitModifier(FINAL)) {
+          final Optional<PsiElement> psiFinal = Arrays.stream(modifierList.getChildren())
+            .filter(psiElement -> FINAL.equals(psiElement.getText()))
+            .findFirst();
+
+          psiFinal.ifPresent(psiElement -> holder.registerProblem(psiElement,
+            "Redundant final field modifier",
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: "field" is wrong here

Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: yep, I've corrected that :)

File: testData/inspection/redundantModifierInspection/ValueClass.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 @lombok.Value
-public class ValueClass {
+public <warning descr="Redundant final field modifier">final</warning> class ValueClass {
```
Comment (mplushnikov) [inline testData/inspection/redundantModifierInspection/ValueClass.java]: You forgot to change your test too :-) It's failing now.

Comment (mplushnikov) [reply testData/inspection/redundantModifierInspection/ValueClass.java]: I'll change it by myself.

**Comment by wyhasany** [comment]
thx @mplushnikov it wasn't the best idea to program in the bus :)

**Pull request merged**